### ISSUE TITLE
Feature request: enable tabbing multiple lines at once in the XML editor

### ISSUE THREAD

**Issue by jaltekruse** [Doenet/DoenetTools#1744]
Keeping XML code organized is hard without being able to move a series of tags in one level of indent as you add a new wrapping tag.

Currently we can indent individual lines, but if you select several lines and hit tab it replaces the selected content with a single tab. I recommend we do something like what is shown in this demo.

 https://codemirror.net/examples/tab/

We are already "breaking" the high level concern that is mentioned on that page about how tab on the web is usually used to move around to different elements with the keyboard, for the sake of navigating to links or form inputs.

I assume we likely want to preserve the behavior we have right now for inserting a single tab anywhere in a line, and I would recommend any time that there is a text selection to switch to the behavior in the demo, to indent at the start of the line, and if a selection spans across multiple lines to indent all lines covered by the selection. I don't think we necessarily want the exact behavior of the demo, where placing your cursor in the middle of the line and then hitting tab will always just indent the start of the line and inserting tabs in the middle of a line is impossible.

**Comment by dqnykamp** [comment]
Yes, since we are breaking the guidelines for accessibility, we need to let folks know how to get out by press Esc+Tab, as they suggest.

If we are enabling tabs, I'm not sure why we shouldn't enable all these features.

The ability to index multiple lines at once is a critical feature.  Thankfully, there is a way even with our current setup: the keys Ctrl+] and Ctrl+[.  I'm OK with using those combinations, now that I discovered them.  But, they aren't as intuitive as the tab key; people probably won't discover them on their own.

**Comment by jaltekruse** [comment]
Bumping this because I was editing some DoenetML and forgot the workaround (but I knew to come here to learn what it was again), I think not having this is pretty annoying for new people learning to use the editor.

**Comment by rogness** [comment]
For what it's worth -- I think lots of people (myself included) are used to Cmd+] and Cmd+[ (or Cntl for the non-mac world) to change indentation of a line or block of lines.  Those shortcuts work throughout the Google-verse (docs, gmail), the LaTeX editor I use, the text editor I use, and so on.  I've been using them in the Doenet editor without even thinking about it, because it's the first key combination my brain thinks to try nowadays when adjusting indents.  I hadn't even checked if the tab key works on blocks of lines.

I certainly wouldn't mind the tab key having that functionality, but would vote to keep the existing shortcuts as well.

Related, but different -- by the end of a long DoenetML document I always find my indentation hasn't been consistent, and it would be nice to have some sort of built-in XML beautifier / HTML tidy feature.  I tried a few online XML beautifiers but they've all conked out because they're not expecting tags like `<math>` or `<graph>`.  (I figured this is so low priority it wasn't worth opening a new issue...)
